pkg/controller/dynamic/dynamic_controller_integration_test.go

```diff
@@ -799,7 +799,11 @@ func getChangedFields(initialObject, updatedObject map[string]interface{}, field
 		for k, v := range updated {
 			if !reflect.DeepEqual(initial[k], v) {
 				if _, ok := v.(map[string]interface{}); ok {
-					changedFields[k] = getChangedFields(initial, updated, k)
+					// Skip checking for changes in resource reference fields, because there
```

Comment:
What if we normalize to "external form" ?

Reply:
In this case, that won't really help. The test checks what is applied (to k8s api) vs the exported resource in gcp. If we normalize internally to use the "external" form, the test will still fail, because it cannot find the "changed" field (`*Ref.name`) in the exported resources (that only have `*Ref.external`).

Reply:
Oh I see, this is our tests!